From 0423906a9de8365ff00d3be3d0e548b7446c2942 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Tue, 23 Jun 2020 19:19:46 +0200
Subject: [PATCH 160/388] ARM: dts: sun8i-a83t-tbs-a711: Give Linux more
 privileges over SCPI

This is needed since on A83T there's no PSCI/ATF implementation.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm/boot/dts/sun8i-a83t.dtsi | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/sun8i-a83t.dtsi b/arch/arm/boot/dts/sun8i-a83t.dtsi
index 7d0bd243b..682fe7005 100644
--- a/arch/arm/boot/dts/sun8i-a83t.dtsi
+++ b/arch/arm/boot/dts/sun8i-a83t.dtsi
@@ -272,7 +272,7 @@ opp-1800000000 {
 
 	scpi_protocol: scpi {
 		compatible = "arm,scpi";
-		mboxes = <&msgbox 2>, <&msgbox 3>;
+		mboxes = <&msgbox 0>, <&msgbox 1>;
 		mbox-names = "tx", "rx";
 		shmem = <&scpi_sram>;
 	};
@@ -410,6 +410,9 @@ syscon: syscon@1c00000 {
 			compatible = "allwinner,sun8i-a83t-system-controller",
 				"syscon";
 			reg = <0x01c00000 0x1000>;
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+                        ranges;
 
 			sram_a2: sram@40000 {
 				compatible = "mmio-sram";
@@ -420,7 +423,7 @@ sram_a2: sram@40000 {
 
 				scpi_sram: scp-shmem@13c00 {
 					compatible = "arm,scp-shmem";
-					reg = <0x13c00 0x200>;
+					reg = <0x13e00 0x200>;
 				};
 			};
 
-- 
2.35.3

